### PR TITLE
Various unit tests, minor fixes, and dead code removal

### PR DESCRIPTION
**Pull request by himanshug**
this patch contains
- Unit Tests for various java classes in druid
- couple of very minor fixes
- removal ot some code that does not get used anywhere

### CODE REVIEW

**Comment by xvrl** [comment]
Can we make the description of this pull-request more specific and give a general indication of the types of tests we are adding?

File: indexing-hadoop/src/main/java/io/druid/indexer/Utils.java [hunk 1]
```diff
@@ -44,30 +40,19 @@
 {
   private static final ObjectMapper jsonMapper = new DefaultObjectMapper();
 
-  public static <K, V> Map<K, V> zipMap(Iterable<K> keys, Iterable<V> values)
```
Comment (xvrl) [inline indexing-hadoop/src/main/java/io/druid/indexer/Utils.java]: any reason we are removing this? This doesn't seem to be part of tests

Comment (himanshug) [inline indexing-hadoop/src/main/java/io/druid/indexer/Utils.java]: thanks, updated the description.

and, this code does not seem to get used anywhere. I can put it back if there is a reason to keep it around.
what do you think?

**Comment by fjy** [comment]
I didn't closely look at every test, but I believe the following is true:
1) No existing tests were changed and the removals are only for dead code
2) The tests run in a reasonable time
3) The tests are increasing code coverage for Druid, which is very important
4) The general code follows the style guide

I am in favor of this PR if the above is true.

File: common/src/main/java/io/druid/collections/IntList.java [hunk 2]
```diff
@@ -112,7 +112,7 @@ public IntBuffer getBaseList(int index)
     final IntBuffer retVal = IntBuffer.wrap(array);
 
     if (index + 1 == baseListCount()) {
-      retVal.limit(maxIndex - (index * allocateSize));
+      retVal.limit(maxIndex - (index * allocateSize) + 1);
```
Comment (drcrallen) [inline common/src/main/java/io/druid/collections/IntList.java]: Why +1?

**Comment by drcrallen** [comment]
There seem to be a few non unit test changes in this PR, can those be separated more clearly somehow?

File: common/src/test/java/io/druid/collections/CombiningIteratorTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    testingIterator.next();
+    EasyMock.verify(peekIterator);
+  }
+
+  @Test(expected = UnsupportedOperationException.class)
+  public void testRemove() throws Exception
+  {
+    testingIterator.remove();
+  }
+}
```
Comment (xvrl) [inline common/src/test/java/io/druid/collections/CombiningIteratorTest.java]: can we add newlines at the end of files?

File: indexing-hadoop/src/test/java/io/druid/indexer/path/GranularityPathSpecTest.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class GranularityPathSpecTest
+{
+  private GranularityPathSpec granularityPathSpec;
+  private final String TEST_STRING_PATH = "TEST";
+  private final String TEST_STRING_PATTERN = "*.TEST";
```
Comment (drcrallen) [inline indexing-hadoop/src/test/java/io/druid/indexer/path/GranularityPathSpecTest.java]: FYI, `setFilePattern` ends up as a parameter to `Pattern.compile` which expects a regex. Please confirm this is actually the regex you want to use

File: common/src/main/java/io/druid/collections/IntList.java (hunk 2 above)
Comment (cheddar) [inline common/src/main/java/io/druid/collections/IntList.java]: It's a fix to a problem found with a unit test.  Check out the branch, remove the +1 and run the tests...

**Comment by drcrallen** [comment]
I'm in agreement with @fjy , but I would really like to see the non-unit tests items in at least a separate commit so its more obvious they are changing.

**Comment by cheddar** [comment]
The non-unit test changes are all fixing bugs that exist in the current code found by the unit tests.  If we remove them, the tests won't pass and then travis will be upset and then what do we do?

**Comment by himanshug** [comment]
@fjy  all of what you said is true.

**Comment by drcrallen** [comment]
@cheddar : Put them in the same PR as per https://github.com/druid-io/druid/pull/1044

File: indexing-hadoop/src/main/java/io/druid/indexer/Utils.java (hunk 1 above)
Comment (xvrl) [inline indexing-hadoop/src/main/java/io/druid/indexer/Utils.java]: I'm fine removing the code if it is not used anywhere.

**Comment by xvrl** [comment]
I'm on board with all those fixes + changes, I updated the PR title to include the fact that we fixed things.
Once you update the license header and squash into either a single commit, or one commit per sub-module I can merge.

File: indexing-hadoop/src/test/java/io/druid/indexer/path/GranularityPathSpecTest.java (hunk 4 above)
Comment (xvrl) [inline indexing-hadoop/src/test/java/io/druid/indexer/path/GranularityPathSpecTest.java]: This appears to be just testing the setter, nothing more, so it should be fine.

Comment (cheddar) [inline indexing-hadoop/src/test/java/io/druid/indexer/path/GranularityPathSpecTest.java]: These tests are pretty innocuous.  All they are doing at this point is verifying the setter, so the pattern provided doesn't really matter...

**Comment by xvrl** [comment]
I'm happy do the squashing and header update for you if that makes it easier.

**Comment by xvrl** [comment]
Merged as part of af807c6215ae5aad

File: common/src/test/java/io/druid/collections/CombiningIteratorTest.java (hunk 3 above)
Comment (himanshug) [inline common/src/test/java/io/druid/collections/CombiningIteratorTest.java]: will do

**Pull request closed without merging**